cli/app.go

```diff
@@ -224,6 +231,49 @@ var dataTagByFilterFlags = append([]cli.Flag{
 },
 	commonFilterFlags...)
 
+func getValFromContext(name string, ctx *cli.Context) interface{} {
+	// some fuzzy searching is required here, because flags are typically in kebab case, but
+	// params are typically in snake or camel case
+	replacer := strings.NewReplacer("_", "-")
+	dashFormattedName := replacer.Replace(strings.ToLower(name))
+
+	value := ctx.Value(dashFormattedName)
+	if value != nil {
+		return value
+	}
+
+	camelFormattedName := matchFirstCap.ReplaceAllString(name, "${1}-${2}")
+	camelFormattedName = matchAllCap.ReplaceAllString(camelFormattedName, "${1}-${2}")
+	camelFormattedName = strings.ToLower(camelFormattedName)
+
+	return ctx.Value(camelFormattedName)
```

Comment:
It seems like some guardrails might be nice here, but I'm not sure that we can say at compile time what's safe and what's not. Even if an argument is non-optional, I expect there are cases where a nil value is normal and expected.